MembraneVideoroomDemo/Controllers/RoomController.swift

```diff
@@ -393,15 +457,54 @@ extension RoomController: MembraneRTCDelegate {
         }
     }
 
-    func onTrackUpdated(ctx _: TrackContext) {}
+    func onTrackUpdated(ctx: TrackContext) {
+        let isActive =
+            (ctx.metadata["active"] != nil)
+            ? ctx.metadata["active"] as! Bool : ctx.metadata["type"] as! String == "camera" ? false : true
+
+        if ctx.metadata["type"] as! String == "camera" {
+            DispatchQueue.main.async {
+                if ctx.peer.id == self.primaryVideo?.participant.id {
+                    self.primaryVideo?.isActive = isActive
+                } else {
+                    self.participantVideos.first(where: { $0.participant.id == ctx.peer.id })?.isActive = isActive
+                }
+                self.objectWillChange.send()
+            }
+        } else {
+            var p = participants[ctx.peer.id]
+            p?.isAudioTrackActive = isActive
+            self.participants[ctx.peer.id] = p
+            DispatchQueue.main.async {
```

Comment:
Take care that `self.participants` and other values related to RoomController are accessed on one thread. Accessing them on two threads can lead to race conditions.

Reply:
Done.